src/prometheus-stack/chart/templates/coredns-netpol.yaml

```diff
@@ -0,0 +1,27 @@
+# Copyright 2025 Defense Unicorns
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+{{ if .Values.addCorednsNetpol }}
```

Comment:
I think this might have been my initial suggestion, but is it better to name this `coreDnsNetpol.enabled` and/or tie it to RKE2 in the name?

Also might be nice to use this in our RKE2 testing instead of the one-off netpol we have now (this would ensure it's tested in CI too).